pkg/services/sqlstore/annotation.go

```diff
@@ -67,10 +70,18 @@ func (r *SqlAnnotationRepo) Update(item *annotations.Item) error {
 		}
 
 		existing.Updated = time.Now().UnixNano() / int64(time.Millisecond)
-		existing.Epoch = item.Epoch
 		existing.Text = item.Text
-		if item.RegionId != 0 {
-			existing.RegionId = item.RegionId
+
+		existing.Epoch = item.Epoch
+		existing.EpochEnd = item.EpochEnd
+
+		if existing.EpochEnd == 0 {
+			existing.EpochEnd = item.Epoch
+		}
+		if existing.EpochEnd > existing.Epoch {
```

Comment:
Hmm, here you have the same logic again as in the API for create/patch annotation. Could this be put in one go package, preferrably only in sqlstore package/this file?

Reply:
good idea.  I removed it from the api, and only have it in sqlstore.  There is now a function that is called for both Save and Update

Reply:
Great